libpromises/evalfunction.c

```diff
@@ -6659,6 +6659,158 @@ static FnCallResult FnCallReadData(ARG_UNUSED EvalContext *ctx,
     return (FnCallResult) { FNCALL_SUCCESS, (Rval) { json, RVAL_TYPE_CONTAINER } };
 }
 
+static int ContainerComparator(JsonElement const *tits,
+                               JsonElement const *ass,
+                               void *user_data)
+{
+    size_t *index = (size_t *)user_data;
+
+    char const *left = JsonPrimitiveGetAsString(JsonAt(tits, *index));
+    char const *right = JsonPrimitiveGetAsString(JsonAt(ass, *index));
+    return strcmp(left, right);
+}
+
+static FnCallResult FnCallClassExprFilter(EvalContext *ctx,
+                                          ARG_UNUSED Policy const *policy,
+                                          FnCall const *fp,
+                                          Rlist const *args)
+{
+    if (args == NULL || args->next == NULL || args->next->next == NULL)
+    {
+        FatalError(ctx, "Function %s requires at least 3 arguments",
+                   fp->name);
+    }
+
+    char const *path = RlistScalarValue(args);
+    args = args->next;
+    size_t const col = IntFromString(RlistScalarValue(args));
```

Comment:
Should maybe be called `class_col` since you also have `sort_col`?

Reply:
Have changed these to `class_index` and `sort_index`.